packages/components/src/tooltip/index.tsx

```diff
@@ -0,0 +1,106 @@
+/**
+ * External dependencies
+ */
+import * as Ariakit from '@ariakit/react/tooltip';
+
+/**
+ * WordPress dependencies
+ */
+import { useInstanceId } from '@wordpress/compose';
+import { Children } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import type { TooltipProps } from './types';
+import Shortcut from '../shortcut';
+import { positionToPlacement } from '../popover/utils';
+
+/**
+ * Time over anchor to wait before showing tooltip
+ *
+ * @type {number}
+ */
+export const TOOLTIP_DELAY = 700;
+
+function Tooltip( props: TooltipProps ) {
+	const { children, delay = TOOLTIP_DELAY, position, shortcut, text } = props;
+
+	const baseId = useInstanceId( Tooltip, 'tooltip' );
+	const describedById = text || shortcut ? baseId : undefined;
+
+	const isOnlyChild = Children.count( children ) === 1;
+	// console error if more than one child element is added
+	if ( ! isOnlyChild ) {
+		if ( 'development' === process.env.NODE_ENV ) {
+			// eslint-disable-next-line no-console
+			console.error(
+				'Tooltip should be called with only a single child element.'
+			);
+		}
+	}
+
+	const DEFAULT_PLACEMENT = 'bottom';
+
+	// Compute tooltip's placement:
+	// - compute it from the legacy `position` prop (if defined)
+	// - otherwise, fallback to the DEFAULT_PLACEMENT.
+	let computedPlacement;
+
+	if ( position !== undefined ) {
+		computedPlacement = positionToPlacement( position );
+	} else {
+		computedPlacement = DEFAULT_PLACEMENT;
+	}
+
+	const tooltipStore = Ariakit.useTooltipStore( {
+		// TODO: can remove the ignore once position has been fully deprecated
+		// Type error due to 'overlay' type from positionToPlacement
+		// @ts-ignore
+		placement: computedPlacement,
```

Comment:
This type error is due to `overlay` from `position`/`Popover`. I thought maybe we could `Omit` it since it isn't exposed in the legacy tooltip API, but I wasn't sure how to execute that.

Reply:
How about:

```diff
diff --git a/packages/components/src/popover/utils.ts b/packages/components/src/popover/utils.ts
index 2112b28596..cf410fbd9a 100644
--- a/packages/components/src/popover/utils.ts
+++ b/packages/components/src/popover/utils.ts
@@ -3,7 +3,7 @@
  */
 // eslint-disable-next-line no-restricted-imports
 import type { MotionProps } from 'framer-motion';
-import type { ReferenceType } from '@floating-ui/react-dom';
+import type { Placement, ReferenceType } from '@floating-ui/react-dom';
 
 /**
  * Internal dependencies
@@ -16,7 +16,7 @@ import type {
 
 const POSITION_TO_PLACEMENT: Record<
 	NonNullable< PopoverProps[ 'position' ] >,
-	NonNullable< PopoverProps[ 'placement' ] >
+	Placement
 > = {
 	bottom: 'bottom',
 	top: 'top',
@@ -79,8 +79,7 @@ const POSITION_TO_PLACEMENT: Record<
  */
 export const positionToPlacement = (
 	position: NonNullable< PopoverProps[ 'position' ] >
-): NonNullable< PopoverProps[ 'placement' ] > =>
-	POSITION_TO_PLACEMENT[ position ] ?? 'bottom';
+) => POSITION_TO_PLACEMENT[ position ] ?? 'bottom';
 
 /**
  * @typedef AnimationOrigin
```

Reply:
Ah, okay. So instead of using the types for `placement` from `Popover`, we are using it from `floating-ui` instead, which doesn't contain `overlay`. 

And `NonNullable` isn't needed anymore?

Reply:
We needed `NonNullable` because the `placement` prop could also be `undefined`. In contrast, the `Placement` type from `floating-ui` does not contain `undefined`, and therefore we don't need to wrap it in `NonNullable` :) 


Once https://github.com/WordPress/gutenberg/pull/54101 is merged, we can rebase / merge trunk to include its changes in this PR :)